src/TransferList/TransferList.tsx

```diff
@@ -155,51 +390,131 @@ export default function TransferList({
-            <Typography data-testid="none-selected" sx={{ pl: 1, pt: 1 }}>
-              None Selected
-            </Typography>
-          ) : (
-            <Box>
-              <Typography sx={{ pl: 1, pt: 1 }}>
-                {`${selectedItems.length} selected`}
-              </Typography>
-              <Button
-                data-testid="clear-all-button"
-                variant="text"
-                sx={{ position: "absolute", right: 2, top: 2 }}
-                onClick={() => onChange([])}
-              >
-                Clear All
-              </Button>
-            </Box>
-          )}
+          <Checkbox
+            aria-label="select all target list items"
+            checked={targetItemsToTransfer.length > 0}
+            disabled={allTargetItems.length === 0}
+            disableRipple
+            indeterminate={
+              targetItemsToTransfer.length > 0 &&
+              allTargetItems.length !== targetItemsToTransfer.length
+            }
+            onClick={handleCheckAllTarget}
+            sx={{ pl: 1 }}
+          />
+          <Box
+            sx={{
+              display: "flex",
+              flexDirection: "column",
+              justifyContent: "center",
+              pl: 1
+            }}
+          >
+            <Typography variant="body1">{targetListLabel}</Typography>
+            <Typography
+              variant="body2"
+              sx={{ color: "text.secondary" }}
+            >{`${targetItemsToTransfer.length}/${allTargetItems.length} selected`}</Typography>
+          </Box>
         </Box>
+
+        {allTargetItems.length > 0 ? (
+          <Box sx={{ px: 3 }}>
+            <Search onChange={setTargetFilter} />
+          </Box>
+        ) : null}
+
         <Box
           sx={{
             height: "100%",
             overflowX: "hidden",
             overflowY: "auto",
-            width: "100%"
+            px: 2
           }}
         >
-          <Box
-            sx={{
-              width: "100%"
-            }}
-          >
-            <DeletableList items={selectedItems} onDelete={handleToggle} />
+          <Box>
+            <SingleList
+              checked={targetItemsToTransfer}
+              items={filteredTargetItems.map(item => ({
+                key: filterKey(item),
+                primaryLabel: getItemLabel(item),
+                secondaryLabel: getItemLabel(item, "secondary")
+              }))}
+              handleToggle={handleToggle}
+              role="target-list"
+            />
           </Box>
         </Box>
       </Box>
     </Box>
   );
 }
+
+const Search = ({ onChange }: { onChange: (value: string) => void }) => {
+  const [search, setSearch] = useState("");
+
+  // handle change
+  const handleChange = (event: {
+    target: {
+      value: string;
+    };
+  }) => {
+    setSearch(event.target.value);
+  };
+
+  // Call on change on every search
+  useEffect(() => {
+    onChange(search);
+  }, [onChange, search]);
```

Comment:
Don't use an effect for this. Call `onChange` in your `handleChange` method.

https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers